spec/helpers/application_helper/buttons/auth_key_pair_cloud_create_spec.rb

```diff
@@ -3,17 +3,17 @@
     it "when the create action is available then the button is not disabled" do
       view_context = setup_view_context_with_sandbox({})
       button = described_class.new(view_context, {}, {}, {})
-      ems = object_double(ManageIQ::Providers::CloudManager.new, :class => ManageIQ::Providers::Openstack::CloudManager)
-      allow(ManageIQ::Providers::Openstack::CloudManager::AuthKeyPair).to receive(:is_available?).and_return(true)
+      ems = object_double(ManageIQ::Providers::CloudManager.new)
+      allow(ems).to receive(:supports?).with(:auth_key_pair_create).and_return(true)
```

Comment:
not a big fan of using doubles in general but that is out of scope.

wonder if does make sense to write a helper method (sorry, can't help it)

```ruby
  def supported_ems(supports)
      ems = object_double(ManageIQ::Providers::CloudManager.new)
      allow(ems).to receive(:supports?).with(:auth_key_pair_create).and_return(supports)
      ems
  end
```

Reply:
@kbrock took your advice and refactored this spec.  See [this commit](https://github.com/ManageIQ/manageiq-ui-classic/pull/7876/commits/4ac1b639b17467bed70e0ce305e60fe5cdc068bb)